stdpopsim/slim_engine.py

```diff
@@ -1085,23 +1082,19 @@ def simulate(
                     "compare results across different values of the scaling factor."
                 )
             )
+        if contig.recombination_map is None:
+            raise ValueError(
+                "Your Contig object is not compatible with the SLiM "
+                "engine. You must specify a recombination map."
+            )
+        if len(contig.genomic_element_types) == 0:
```

Comment:
I think this behavior should be both documented and possibly emit a warning.

Reply:
I think it might be simpler if the `Contig` class just create a flat recombination map when non is specified.